pre_commit_hooks/destroyed_symlinks.py

```diff
@@ -0,0 +1,108 @@
+import argparse
+import sys
+from operator import methodcaller
+from subprocess import check_call
+from subprocess import check_output
+from typing import Optional
+from typing import Sequence
+
+ORDINARY_CHANGED_ENTRIES_MARKER = b'1'
+PERMS_LINK = b'120000'
+PERMS_NONEXIST = b'000000'
+
+
+def normalize_content(content: bytes) -> bytes:
+    return b'\n'.join(
+        filter(
+            None,
+            map(
+                methodcaller('strip'),
+                content.splitlines(),
+            ),
+        ),
+    )
+
+
+def find_destroyed_symlinks(autofix: bool) -> Sequence[bytes]:
+    destroyed_links = []
+    for line in check_output(
+        ['git', 'status', '--porcelain=v2', '-z'],
+    ).split(b'\0'):
```

Comment:
there's a `zsplit` function that might be helpful, not sure if there's a copy here in `pre_commit_hooks/util.py` -- if not take the one from https://github.com/pre-commit/pre-commit/blob/3fa9d3d7588fa3aee37e04799ad36d2a0fe32e5a/pre_commit/git.py#L19-L24

that should remove the need for checking `if splitted` below

actually, looks like there's an implementation in `check-executables-have-shebangs` -- please move that into the util module and it's tests (if applicable)

Reply:
Initially, I used `bytes` everywhere, so I've added support for `bytes` to `zsplit()`. However, after I've started using filenames passed as arguments, It became more tricky because `sys.argv` is `List[str]`. So, I gave up on using `bytes` but didn't removed the support from `zsplit()` — may be it would be useful someday.

Reply:
"might be used one day" is known as code debt and maintenance burden -- please revert that if it's not being used

also a function taking both `str` and `bytes` is a wart, there should be `zsplit` and `zsplit_b` as separate functions

Reply:
> also a function taking both str and bytes is a wart, there should be zsplit and zsplit_b as separate functions

Yes and no. If a function would automatically convert one to another, try to guess encoding (or assume ASCII) and so on, basically everything what was common in python2, then yes, that's not good.
In my case, it's more like a generic function: if you pass it bytes, it will return list of bytes and this even checked statically by mypy and various language servers so you won't mess one with another.

Anyway, I agree about dead code so I've removed it.

Reply:
mmmm, yes and no but mostly no -- your function wasn't generic, [it manually checked the types](https://github.com/pre-commit/pre-commit-hooks/commit/82fe00463b4265fe2c419ba65ae645a2a647310d#) 😉